eventauth.go

```diff
@@ -632,7 +636,15 @@ func redactEventAllowed(event Event, authEvents AuthEventProvider) error {
 		return err
 	}
 
-	senderDomain, err := domainFromID(event.Sender())
+	roomVersion := allower.create.RoomVersion
+	if roomVersion != nil && *roomVersion != "1" && *roomVersion != "2" {
```

Comment:
I assume we should start defining these as constants at some point, but that can wait until full room version support is implemented.